guides/release/templates/handlebars-basics.md

```diff
@@ -1,85 +1,203 @@
+```
+
+## Things you might see in a Template
+
+A lot more than just HTML markup can go in Templates. In the other pages of this guide, we will cover the features one at a time. In general, special Ember functionality will appear inside curly braces, like this: `{{example}}`. Here are a few examples of Ember Handlebars in action:
+
+Route example:
+```hbs {data-filename=app/templates/application.hbs data-update=true}
+{{!-- Example of a comment that will be invisible --}}
+
+{{!-- outlet determines where a child route's content should render. Don't delete it until you know more about it! --}}
+<div>
+  {{outlet}}
+</div>
+
+{{!-- One way to use a component within a template. This is called "Angle Bracket Component syntax." --}}
+<MyComponent />
+
+```
+
+Component example:
+```hbs {data-filename=app/components/templates/my-component.hbs data-update=true}
+{{!-- An attribute that is defined in a component's JavaScript file --}}
+{{this.numberOfSquirrels}}
+
+{{!-- Some data passed down from a parent component or controller --}}
+{{weatherStatus}}
+
+{{!-- This button uses Ember Actions to make it interactive. A method named `plantATree` is called when the button is clicked --}}
+<button onclick={{action 'plantATree'}}>
+  More trees!
+<button>
+
+{{!-- Here's an example of template logic in action. If the `this.skyIsBlue` attribute is `true`, the text inside will be shown --}}
+{{#if this.skyIsBlue}}
+  If the skyIsBlue attribute is true, show this message
+{{/if}}
+
+{{!-- You can pass a whole block of markup and handlebars content from one component to another. yield is where the block shows up when the page is rendered --}}
+{{yield}}
+```
+
+Lastly, it's important to know that data can be passed from
+one Template to another. Here's what it looks like
+to pass data from a Route Template to a Component Template.
+
+```handlebars {data-filename=app/templates/application.hbs}
+<MyComponent @favoriteFramework=this.favoriteFramework />
+```
+
+Learn more about passing data between Templates in [../../components/passing-properties-to-a-component].
+
+## Helper functions
+
+Ember Helpers are pure functions that can be used in any template to change how data is displayed. For example, you could write a Helper function that capitalizes a word, does some math, converts a currency, or more.
+A Helper takes in `parameters`, which is an array of the values passed into the function, and should return a value.
+Ember gives you the ability to [write your own helpers](../writing-helpers/), and comes with some [helpers built-in]().
+
+For example, let's say you would like the ability to add two numbers together.
+Define a function in `app/helpers/sum.js` to create a `sum` helper:
 
-For example, let's say you would like the ability to add a few numbers together, without needing to define a computed property everywhere you would like to do so.
 
 ```javascript {data-filename=app/helpers/sum.js}
 import { helper } from '@ember/component/helper';
```

Comment:
The helper API has changed, I commented at https://github.com/ember-learn/guides-source/issues/460#issuecomment-464419841

```
import { helper as buildHelper } from '@ember/component/helper';

....

export const helper = buildHelper(helloHelper);
```